---
title: "EVscope output report"
author:
  - Rambo (Yiyong) Zhao^[yiyong.zhao@yale.edu]
  - Xianjun Dong^[xianjun.dong@yale.edu]
  - Himanshu Chintalapudi^[himanshu.chintalapudi@yale.edu]
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::html_document2:
    collapsed: true
    fig_width: 8
    highlight: haddock
    number_sections: false
    keep_md: false
    theme: cerulean
    toc: true
    toc_depth: 4
    toc_float:
      collapsed: yes
editor_options:
  chunk_output_type: console
params:
  output_dir: NULL
---

<style type="text/css">
.container-fluid {
  max-width: 100% !important;
  padding-left: 10px;
  padding-right: 10px;
}

body, p {
  font-family: "Calibri";
  color: #333333;
}

h1.title {
  font-family: "Marcellus";
  color: dodgerblue4;
  font-size: 50px;
}

h1,h2,h3,h4,h5,h6 {
  font-family: "Marcellus";
  color: dodgerblue4;
}

.email {
  color: #333333;
}

.list-group-item{
  font-family: "Marcellus";
  font-size: 20px;
}

.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
  background-color: #00BFFF;
}

.btn-rcode {
  background-color: #00BFFF;
  color: white;
}

#TOC {
  max-width: fit-content;
  white-space: nowrap;
  max-height: calc(100vh - 50px);
}
</style>

```{r setup, echo=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = FALSE, fig.height=8, fig.width=8, message=FALSE, warning=FALSE, cache=FALSE, fig.align = 'center')

suppressWarnings(suppressMessages({
  library(DT)
  library(kableExtra)
  library(here)
}))

downloadableDT <- function(atable,
                         rownames = NULL,
                         pageLength = 50,
                         colnames = NULL,
                         options = list()){
  require(DT)
  
  # 确保输入是数据框
  if(!is.data.frame(atable)) {
    atable <- as.data.frame(atable)
  }
  
  # 确保列名有效
  if(is.null(colnames)) {
    colnames <- colnames(atable)
  }
  
  # 构建选项列表
  dt_options <- list(
    autoWidth = TRUE,
    scrollX = FALSE,
    dom = 'Blfrtip',
    pageLength = pageLength,
    buttons = c('copy', 'csv', 'excel'),
    lengthMenu = list(c(10, 20, -1), c('10', '20', 'All')),
    columnDefs = list(
      list(targets = seq_len(ncol(atable)) - 1, className = "dt-left")
    )
  )
  
  # 合并用户提供的选项
  if(length(options) > 0) {
    dt_options <- modifyList(dt_options, options)
  }
  
  # 创建数据表
  datatable(
    atable,
    rownames = rownames,
    colnames = colnames,
    extensions = 'Buttons',
    options = dt_options,
    class = "compact stripe hover"
  )
}


```
**This report contains a comprehensive breakdown of all relevant QC reports, log files and plots for all steps after running EVscope.  **

```{r, echo=FALSE, results='asis'}
knitr::asis_output("\n\n&nbsp;\n\n")
```

```{r, out.width="75%", out.height="75%"}
knitr::include_graphics("/home/yz2474/scripts/donglab/EVscope/figures/EVscope_pipeline.png")
```

```{r, echo=FALSE, results='asis'}
fp = file.path

here::i_am(knitr::current_input()) 
proj_dir<- here()


output_dir <- params$output_dir

if (is.null(output_dir) || !dir.exists(output_dir)) {
    stop("Error: The output directory was not found, required module outputs have not been generated by the pipeline!")
}

```

# Raw_data_fastqc {.tabset}

This section contains the QC reports generated by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc">FastQC</a>.

```{r, echo=FALSE, results='asis'}
raw_fastc_01<- list.files(fp(output_dir, "raw_data_fastqc"), pattern = ".html", full.names = T)

html_links <- paste0("[", basename(raw_fastc_01), "](", raw_fastc_01, ")")

for (file in raw_fastc_01) {
  cat(basename(file), "\n\n")  # Display the filename
  cat(sprintf('<iframe src="%s" width="100%%" height="1000px"></iframe>\n\n', file))  # Embed the file in an iframe
}

cat('<br>')
cat('<br>')
```

# UMI_motif


This section contains the results of UMI motif extraction (14-bp motifs) from Read2

```{r}
umi_motif_files<- list.files(fp(output_dir, "02_UMI_motif"), full.names = T)
umi_motif_png<- umi_motif_files[grepl("png", umi_motif_files)]
knitr::include_graphics(umi_motif_png)

umi_motif_csv_name<- umi_motif_files[grepl(".csv", umi_motif_files)]
```

`r basename(umi_motif_csv_name)`

```{r, class.output="scroll-100"}
umi_motif_02_csv<- read.csv(umi_motif_csv_name)

#downloadableDT(umi_motif_02_csv, pageLength=10)
datatable(umi_motif_02_csv, rownames = NULL, options = list(scrollX=F))
```


```{r, results='asis'}
cat("<br><br><br>") 
```

# Trim_galore

```{r, echo=FALSE, results='asis',  class.output="scroll-100"}
trim_galore_03_files<- list.files(fp(output_dir, "03_trim_galore"), full.names = T)

trim_galore_03_reports<- trim_galore_03_files[grepl("report", trim_galore_03_files)]

for (file in trim_galore_03_reports) {
  cat(basename(file), "\n\n")  # Display the filename
  cat(sprintf('<iframe src="%s" width="100%%" height="600px"></iframe>\n\n', file))  # Embed the file in an iframe
}

cat('<br>')
cat('<br>')
```

# Fastqc_after_trim-galore 

```{r, echo=FALSE, results='asis'}
trim_galore_fastqc_files<- list.files(fp(output_dir, "04_trim_galore_fastqc"), full.names = T, pattern = ".html")

for (file in trim_galore_fastqc_files) {
  cat(basename(file), "\n\n")  # Display the filename
  cat(sprintf('<iframe src="%s" width="100%%" height="1000px"></iframe>\n\n', file))  # Embed the file in an iframe
}

cat('<br>')
cat('<br>')
```

# Read_strand_detection

```{r}
RNA_strand_files<- list.files(fp(output_dir, "07_strand_detection"), full.names = T)

RNA_strand_table<- RNA_strand_files[grepl("bam2strandness.tsv", RNA_strand_files)]

```

`r basename(RNA_strand_table)`

```{r, echo=FALSE, results='asis'}
RNA_strand_tsv<- read.table(RNA_strand_table, header = TRUE, sep="\t")

datatable(
  RNA_strand_tsv,
  rownames = FALSE,
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel'),
    pageLength = 10,
    columnDefs = list(
      list(targets = "_all", className = "dt-left")
    )
  )
)

RNA_strand_pngs<- RNA_strand_files[grepl("png", RNA_strand_files)]


for (png in RNA_strand_pngs) {
  cat(basename(png), "<br>\n")
  cat(basename(png), "<br>\n")
  cat(sprintf('<img src="%s" style="max-width:100%%; height:auto;"><br><br>\n\n', png))
}

cat('<br>')
cat('<br>')
```


# CIRCexplorer2

```{r,  class.output="scroll-100"}
CIRCexplorer2_files<- list.files(fp(output_dir, "08_circRNA_detection/CIRCexplorer2"), full.names = T)

CIRCexplorer2_cpm_tsv<- CIRCexplorer2_files[grepl("\\_CPM.tsv$", CIRCexplorer2_files)]
```

`r basename(CIRCexplorer2_cpm_tsv)`

```{r,  class.output="scroll-100"}
CIRCexplorer2_cpm_tsv<- read.table(CIRCexplorer2_cpm_tsv, header = T)

downloadableDT(CIRCexplorer2_cpm_tsv)

CIRCexplorer2_circularRNA<- CIRCexplorer2_files[grepl("circularRNA", CIRCexplorer2_files)]
```

`r basename(CIRCexplorer2_circularRNA)`

```{r}
CIRCexplorer2_circularRNA<- read.table(CIRCexplorer2_circularRNA)

downloadableDT(CIRCexplorer2_circularRNA)
```

```{r, results='asis'}
cat("<br><br><br>") 
```

# CIRI2

```{r}
BWA_CIRI2_files<- list.files(fp(output_dir, "08_circRNA_detection/BWA_CIRI2"), full.names = T)

BWA_CIRI2_08_readcounts_CPM <- BWA_CIRI2_files[grepl("\\_readcounts_CPM.tsv$", BWA_CIRI2_files)]

```

`r basename(BWA_CIRI2_08_readcounts_CPM)`

```{r}
BWA_CIRI2_08_readcounts_CPM<- read.table(BWA_CIRI2_08_readcounts_CPM, header = T)

downloadableDT(BWA_CIRI2_08_readcounts_CPM)

BWA_CIRI2_CIRI2_out<- BWA_CIRI2_files[grepl("\\_CIRI2_out.tsv$", BWA_CIRI2_files)]
```

`r basename(BWA_CIRI2_CIRI2_out)`

```{r}
BWA_CIRI2_CIRI2_out<- read.delim(BWA_CIRI2_CIRI2_out)

downloadableDT(BWA_CIRI2_CIRI2_out)
```

```{r, results='asis'}
cat("<br><br><br>") 
```

# Merge_CIRI2_CIRCexplorer2

```{r}
merge_CIRI2_CIRCexplorer2_files<- list.files(fp(output_dir, "08_circRNA_detection/merge_CIRI2_CIRCexplorer2"), full.names = T)

merge_CIRI2_CIRCexplorer2_CIRI2_tsv<- merge_CIRI2_CIRCexplorer2_files[grepl("\\CIRI2.tsv$", merge_CIRI2_CIRCexplorer2_files)]

```

`r basename(merge_CIRI2_CIRCexplorer2_CIRI2_tsv)`

```{r}
merge_CIRI2_CIRCexplorer2_CIRI2_tsv<- read.delim(merge_CIRI2_CIRCexplorer2_CIRI2_tsv)

downloadableDT(merge_CIRI2_CIRCexplorer2_CIRI2_tsv)
```


```{r,echo=FALSE, results='asis'}
merge_CIRI2_CIRCexplorer2_pngs<- merge_CIRI2_CIRCexplorer2_files[grepl("\\.png$", merge_CIRI2_CIRCexplorer2_files)]

for (png in merge_CIRI2_CIRCexplorer2_pngs) {
 cat(basename(png), "<br>\n")
 cat(sprintf('<img src="%s" style="max-width:100%%; height:auto;"><br><br>\n\n', png))
}

cat('<br>')
cat('<br>')
```

# Picard_metrics

```{r, fig.height=12}
picard_metrics_files<- list.files(fp(output_dir, "09_picard_metrics"), full.names = T)
picard_metrics_pngs<- picard_metrics_files[grepl(".png", picard_metrics_files)]
for (png in picard_metrics_pngs) {
  cat(basename(png), "<br>\n")
  cat(sprintf('<img src="%s" style="max-width:100%%; height:auto;"><br><br>\n\n', png))
}
```

```{r, results='asis'}
cat("<br><br><br>") 
```


# RNA_distribution

```{r}
plot_RNA_distribution_files<- list.files(fp(output_dir, "11_RNA_distribution/featureCounts"), full.names = T)

plot_RNA_distribution_tsv<- plot_RNA_distribution_files[grepl("CPM.tsv", plot_RNA_distribution_files)]

```

`r basename(plot_RNA_distribution_tsv)`

```{r, echo=FALSE, results='asis'}
plot_RNA_distribution_tsv<- read.table(plot_RNA_distribution_tsv, header = T)

downloadableDT(plot_RNA_distribution_tsv)

plot_RNA_distribution_pngs<- plot_RNA_distribution_files[grepl("png", plot_RNA_distribution_files)]


for (png in plot_RNA_distribution_pngs) {
  cat(basename(png), "<br>\n")
  cat(sprintf('<img src="%s" style="max-width:100%%; height:auto;"><br><br>\n\n', png))
}


cat('<br>')
cat('<br>')
```

# Contamination_check

```{r}
kraken_15_files<- list.files(fp(output_dir, "13_Kraken"), full.names = T, pattern = ".html")
```

`r basename(kraken_15_files)`

<iframe src=`r kraken_15_files` width="100%" height="600px"></iframe>


```{r, results='asis'}
cat("<br><br><br>") 
```

# Reads_mapping_stats

```{r}
reads_mapping_stats_files<- list.files(fp(output_dir, "12_reads_mapping_stats"), full.names = T)
reads_mapping_stats_png<- reads_mapping_stats_files[grepl(".png", reads_mapping_stats_files)]

```

`r basename(reads_mapping_stats_png)`

```{r, echo=FALSE, results='asis'}
for (png in reads_mapping_stats_png) {
  cat(basename(png), "<br>\n")
  cat(sprintf('<img src="%s" style="max-width:100%%; height:auto;"><br><br>\n\n', png))
}

cat('<br>')
cat('<br>')
```


# QC_matrix_generation

```{r}
QC_matrix_files <- list.files(fp(output_dir, "16_QC_matrix_generation"), full.names = TRUE, pattern = "\\.tsv$")

QC_matrix_tsv <- read.table(QC_matrix_files, 
                            header = TRUE, 
                            sep = "\t", 
                            quote = "",
                            check.names = FALSE)
```

`r basename(QC_matrix_files)`

```{r}
downloadableDT(QC_matrix_tsv, rownames = FALSE, 
               colnames = c("Metric", "Value"))
```

```{r, results='asis'}
cat("<br><br><br>") 
```

# R session info
This section includes technical details about the R packages used to produce this report. Most readers can safely ignore it.

<button class="btn btn-rcode" data-toggle="collapse" data-target="#BlockName"> Show/Hide R Session Details </button>
<div id="BlockName" class="collapse">

```{r session info}
sess <- devtools::session_info()
sess$platform
knitr::kable(as.data.frame(sess$packages), caption="Package versions")
```

</div>

<div class="tocify-extend-page" data-unique="tocify-extend-page" style="height: 0;"></div>

